src/Umbraco.Core/Services/UserService.cs

```diff
@@ -912,13 +912,24 @@ private async Task<UserOperationStatus> ValidateUserCreateModel(UserCreateModel
             return Attempt.FailWithStatus<IUser?, UserOperationStatus>(UserOperationStatus.MissingUser, existingUser);
         }
 
-        var userGroups = _userGroupRepository.GetMany().Where(x=>model.UserGroupKeys.Contains(x.Key)).ToHashSet();
+        IEnumerable<IUserGroup> allUserGroups = _userGroupRepository.GetMany().ToArray();
```

Comment:
I know most of the returns in this method is not new in this PR, but same comment as before. We need to complete the scope, if nothing should be rolled back.

Reply:
Very good point 😄 I've completed the scopes where it's safe to do so in the 2 methods 😄